packages/flame/lib/src/game/game_widget/game_widget.dart

```diff
@@ -210,34 +165,17 @@ class _GameWidgetState<T extends Game> extends State<GameWidget<T>> {
   @override
   void dispose() {
     super.dispose();
+    widget.game.removeGameStateListener(onGameStateChange);
     widget.game.onRemove();
-    removeOverlaysListener(widget.game);
     // If we received a focus node from the user, they are responsible
     // for disposing it
     if (widget.focusNode == null) {
       _focusNode.dispose();
     }
   }
 
-  void addMouseCursorListener() {
-    widget.game.mouseCursor.addListener(onChangeMouseCursor);
-  }
-
-  void onChangeMouseCursor() {
-    setState(() {
-      _mouseCursor = widget.game.mouseCursor.value;
-    });
-  }
-
-  //#region Widget overlay methods
-
-  void addOverlaysListener() {
-    widget.game.overlays.addListener(onChangeActiveOverlays);
-    initialActiveOverlays = widget.game.overlays.value;
-  }
-
-  void removeOverlaysListener(T game) {
-    game.overlays.removeListener(onChangeActiveOverlays);
+  void onGameStateChange() {
+    setState(() {});
```

Comment:
Hmm... Is this really kosher?

Reply:
What `setState()` does is, removing all the asserts:
```dart
fn();
_element!.markNeedsBuild();
```
so passing a no-op function seems fine to me, considering that all we want to do is to schedule a rebuild of the `GameWidget`.